src/timearray.jl

```diff
@@ -259,63 +261,111 @@ function print_time_array(io::IO, ta::TimeArray{T}, short=false) where T
-                i = size(strs, 1) - i
-                println(io)
-                print(io, "│ ", ts[i], " ")
-                for j in p
-                    print(io, "│ ", _rpad(strs[i, j], colwidth[j] + 1))
-                end
-                print(io, "│")
+Helper function to print a single page of the `TimeArray`
+"""
+function _print_page(io::IO, p::UnitRange{Int}, ta::TimeArray, spacetime, 
+    colwidth, nrow, drow, res_row, tophalf, bothalf, pages) 
+    
+    strs = _showval.(values(ta))
+    ts = timestamp(ta)
+    last = pages > 1 ? "│\u22EF" : "│"
+
+    # row label line
+    ## e.g. | Open  | High  | Low   | Close  |
+    print(io, "│", " "^(spacetime + 2))
+    for (name, w) in zip(colnames(ta)[p], colwidth[p])
+        print(io, "│ ", _rpad(name, w + 1))
+    end
+    println(io, last)
+
+    ## e.g. ├───────┼───────┼───────┼────────┤
+    print(io, "├", "─"^(spacetime + 2))
+    for w in colwidth[p]
+        print(io, "┼", "─"^(w + 2))
+    end
+    print(io, "┤")
+
+    # timestamp and values line
+    if nrow > (drow - res_row)
+        
+        # print bottom part
+        for i in tophalf
+            println(io)
+            print(io, "│ ", ts[i], " ")
+            for j in p
+                print(io, "│ ", _rpad(strs[i, j], colwidth[j] + 1))
             end
+            print(io, i % 3 == 0 ? last : "│")
+        end
 
-        else
-            for i in 1:nrow
-                println(io)
-                print(io, "│ ", ts[i], " ")
-                for j in p
-                    print(io, "│ ", _rpad(strs[i, j], colwidth[j] + 1))
-                end
-                print(io, "│")
+        # print vdots part
+        println(io)
+        print(io, "│ ", _rpad("\u22EE", spacetime + 1))
+        for j in p
+            print(io, "│ ", _rpad("\u22EE", colwidth[j] + 1))
+        end
+        # print(io, pages > 1 ? "│\u22F1" : "│")
+        print(io, "│")
+
+        # print bottom part
+        for i in (length(bothalf) - 1):-1:0
+            i = size(strs, 1) - i
+            println(io)
+            print(io, "│ ", ts[i], " ")
+            for j in p
+                print(io, "│ ", _rpad(strs[i, j], colwidth[j] + 1))
             end
+            print(io, i % 3 == 0 ? last : "│")
         end
 
-        if length(pages) > 1 && p != pages[end]
-            print(io, "\n\n")
+    else
+        # print all rows
+        for i in 1:nrow
+            println(io)
+            print(io, "│ ", ts[i], " ")
+            for j in p
+                print(io, "│ ", _rpad(strs[i, j], colwidth[j] + 1))
+            end
+            print(io, i % 3 == 0 ? last : "│")
         end
-    end  # for p ∈ pages
+    end
 end
-Base.show(io::IO, ta::TimeArray) = print_time_array(io, ta, true)
-Base.show(io::IO, ::MIME"text/plain", ta::TimeArray) =
-    print_time_array(io, ta, false)
+
+Base.show(io::IO, ta::TimeArray) = 
+    print_time_array(io, ta, get(io, :limit, false), get(io, :limit, true))
```

Comment:
I found that `summary` is a more proper method for the setting `short=true`.

So how about this?
```suggestion
Base.summary(io::IO, ta::TimeArray) = print_time_array(io, ta, true)

Base.show(io::IO, ta::TimeArray) = 
    print_time_array(io, ta, false, get(io, :limit, true))
```